bot/cogs/image_dots_cog.py

```diff
@@ -0,0 +1,202 @@
+        # if you want an inverted image, you flip dots >= threshold to dots < threshold
+        # or simply not(...)
+        # or the bitwise not operator on the result
+        for i in range(len(dots)):
+            dots[i] = chr(ord('1') - inverted) if dots[i] >= threshold else chr(ord('0') + inverted)
+        # now we do some more vodoo magic
+        # actually quite clever technique of binary representation of braille
+        # again, see https://en.wikipedia.org/wiki/Braille_Patterns
+        # braille starts at hex(0x2800) or int(10240)
+        dots.reverse()
+        return str(chr(0x2800 + int(''.join(dots),2)))
+
+    def parse_image(self, image, asciiWidth, threshold, inverted):
+        # necessary to have a gray scaled image. Can do this with a conversion
+        # or simply finding the average of (r,g,b) per pixel value
+        rgb_pixels = image.convert('L')
+        width, height = rgb_pixels.size
+        
+        # new image height. Want to proportionately make the image fit the screen
+        # not sure the exact conversion calculation here.
+        asciiHeight = math.ceil( asciiWidth * ASCIIXDOTS * ( height / width ) / ASCIIYDOTS )
+        # want equal number of asciiDots. i.e. 4x2 array
+        width = asciiWidth * ASCIIXDOTS
+        height = asciiHeight * ASCIIYDOTS
+        rgb_pixels = rgb_pixels.resize((width,height)).getdata()
+        # conversion of a 1d array to 2d array
+        two_d_array = self.to_matrix(list(rgb_pixels), width)
+        
+        # fix the dimensions to not exceed Discord's character limit
+        while (width * height) / (ASCIIXDOTS * ASCIIYDOTS) > CHAR_LIMIT:
+            width -= ASCIIXDOTS
+            height -= ASCIIYDOTS
+
+        # now time for the magic
+        finished_image = []
+        
+        # want to make small subsets of the image. i.e. section the image in asciiHeight*asciiWidth
+        # sections.
+        # . .
+        # . .
+        # . .
+        # . .
+        for y in range(0,height, ASCIIYDOTS):
+            line_of_braille = ''
+            for x in range(0,width,ASCIIXDOTS):
+                # want to get that subsection now. starting in corner (x,y) to (width,height).
+                # in our case we want ASCIIXDOTS width, and ASCIIYDOTS height
+                # referencing: https://stackoverflow.com/questions/38049214/how-to-obtain-a-subarray-in-python-3
+                # we want to get a subset of the image. We can easily use a package like numpy, but
+                # relying on numpy makes the project dependencies huge.
+                line_of_braille += self.image_data_to_braille([sub[x:x+ASCIIXDOTS] for sub in two_d_array[y:y+ASCIIYDOTS]], inverted, threshold)
+
+            finished_image.append(line_of_braille)
+        return finished_image
+
+    async def todots_helper(self, ctx, image, device = None, threshold = 150, inverted = 0) -> None:
+        filename = image
+        if inverted not in (0, 1):
```

Comment:
Inverted is a Boolean remember, we don't need to check it cuz the converter handles that for us